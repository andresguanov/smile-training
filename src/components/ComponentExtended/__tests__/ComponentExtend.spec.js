import { mount } from "@vue/test-utils"
import smComponentExtended from "@components/ComponentExtended/ComponentExtended"
import smButton from "@components/Button/Button"
let wrapper

describe("smComponentExtended", () => {
  //#region Basic render
  it("renders button componentExtend", async () => {
    const buttonClass = "button-class"
    const style = "padding: 16px;"
    const props = {
      label: "label",
      size: "small"
    }
    const click = jest.fn()
    wrapper = mount(smComponentExtended, {
      attrs: {
        class: buttonClass,
        style,
        props,
        on: {
          click
        }
      },
      propsData: {
        component: smButton
      }
    })
    expect(wrapper.findComponent(smButton).exists()).toBe(true)
    expect(wrapper.findComponent(smButton).vm.$options.propsData).toMatchObject(props)
    await wrapper.findComponent(smButton).trigger("click")
    expect(click).toHaveBeenCalledTimes(1)
    expect(wrapper.html()).toMatchSnapshot()
  })
  //#endregion

  //#region Props
  //#endregion

  //#region Methods
  //#endregion

  //#region Events
  //#endregion

  //#region Slots
  //#endregion
})
