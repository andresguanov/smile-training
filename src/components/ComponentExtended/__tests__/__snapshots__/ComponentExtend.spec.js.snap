// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smComponentExtended renders button componentExtend 1`] = `
<button type="button" class="button is-basic is-small button-class" style="padding: 16px;">
  <!----> label
  <!---->
  <!----></button>
`;
